docs/ABI/Mangling.rst

```diff
@@ -216,19 +216,28 @@ Entities
   ADDRESSOR-KIND ::= 'p'                     // pinning addressor (native owner)
 
   decl-name ::= identifier
-  decl-name ::= identifier 'L' INDEX         // locally-discriminated declaration
-  decl-name ::= identifier identifier 'LL'   // file-discriminated declaration
+  decl-name ::= identifier 'L' INDEX                  // locally-discriminated declaration
+  decl-name ::= identifier identifier 'LL'            // file-discriminated declaration
+  decl-name ::= identifier 'L' RELATED-DISCRIMINATOR  // related declaration
+
+  RELATED-DISCRIMINATOR ::= [a-j]
+  RELATED-DISCRIMINATOR ::= [A-J]
 
   file-discriminator ::= identifier 'Ll'     // anonymous file-discriminated declaration
 
-The identifier in a ``<file-discriminator>`` and the first identifier in a
+The identifier in a ``<file-discriminator>`` and the second identifier in a
 file-discriminated ``<decl-name>`` is a string that represents the file the
 original declaration came from. It should be considered unique within the
-enclosing module. The second identifier is the name of the entity. Not all
+enclosing module. The first identifier is the name of the entity. Not all
 declarations marked ``private`` declarations will use this mangling; if the
 entity's context is enough to uniquely identify the entity, the simple
 ``identifier`` form is preferred.
 
+Twenty operators of the form 'LA', 'LB', etc. are reserved to described
+entities related to the entity whose name is provided. For example, 'LE' and
+'Le' in the "SC" module are used to represent the structs synthesized by the
+Clang importer for various "error code" enums.
```

Comment:
Why both?

Reply:
Enums in the tag namespace vs. anonymous enums with a typedef-for-linkage, to catch the case where someone obnoxiously defines the same enum in both namespaces. But that's not part of the ABI.